### PR TITLE
Replace cgmath bindings by `mint`

### LINKED ISSUE #1290: Replace cgmath bindings by `mint`
https://github.com/kvark/mint

cc @ebkalderon 

Also need to cleanup the dependencies. I.e. instead of having this in the root Cargo.toml:
```toml
cgmath-types = ["gfx/cgmath-types", "gfx_core/cgmath-types"]
```
We could just have this:
```toml
mint = "gfx/mint"
```
Which would then require "gfx_core/mint" internally.

### PR DESCRIPTION
**Pull request by vitvakatu**
Closes #1290 

Unfortunately (or maybe not), `Rad` and `Deg` types were removed.

### CODE REVIEW

File: src/core/Cargo.toml [hunk 1]
```diff
@@ -29,14 +29,13 @@ path = "src/lib.rs"
 
 [dependencies]
 bitflags = "0.8"
-cgmath = { version = "0.14", optional = true }
+mint = { version = "0.3", optional = true }
```
Comment (kvark) [inline src/core/Cargo.toml]: please use "0.4.1" here

File: src/core/src/shade.rs [hunk 2]
```diff
@@ -21,8 +21,13 @@ use std::error::Error;
 use {Resources};
 use {AttributeSlot, ColorSlot, ConstantBufferSlot, ResourceViewSlot, SamplerSlot, UnorderedViewSlot};
 
-#[cfg(feature = "cgmath-types")]
-use cgmath::{Deg, Matrix2, Matrix3, Matrix4, Point2, Point3, Rad, Vector2, Vector3, Vector4};
+#[cfg(feature = "mint")]
+use mint::{
+    Point2, Point3,
```
Comment (kvark) [inline src/core/src/shade.rs]: I don't think it makes sense to use all the types explicitly as opposed to a simple `use mint;`, given that all the usage is in macros, so there are no ergonomics concerns

File: src/render/Cargo.toml [hunk 3]
```diff
@@ -30,12 +30,11 @@ name = "gfx"
 path = "src/lib.rs"
 
 [features]
-cgmath-types = ["gfx_core/cgmath-types", "cgmath"]
 serialize = ["gfx_core/serialize", "draw_state/serialize"]
 unstable = []
 
 [dependencies]
-cgmath = { version = "0.14", optional = true }
+mint = { version = "0.3", optional = true }
```
Comment (kvark) [inline src/render/Cargo.toml]: same, please use "0.4.1"

File: src/render/src/shade.rs [hunk 4]
```diff
@@ -14,8 +14,12 @@
 
 //! Shader parameter handling.
 
-#[cfg(feature = "cgmath-types")]
-use cgmath::{Deg, Matrix2, Matrix3, Matrix4, Rad, Point2, Point3, Vector2, Vector3, Vector4};
+#[cfg(feature = "mint")]
+use mint::{
+    Point2, Point3,
```
Comment (kvark) [inline src/render/src/shade.rs]: similarly, please just do `use mint`

**Comment by vitvakatu** [comment]
Appveyor nightly fails because of https://github.com/rust-lang/rust/issues/42544
I had the same problem on my local machine

**Pull request merged**